Tools/TestWebKitAPI/Tests/WTF/WorkQueue.cpp

```diff
@@ -354,4 +354,17 @@ TEST(WTF_WorkQueue, ThreadSafetyAnalysisAssertIsCurrentWorks)
         EXPECT_EQ(iterationCount + 1, holder.result);
 }
 
+#if ASSERT_ENABLED
+TEST(WTF_WorkQueue, MainWorkQueueIsCurrent)
+{
+    assertIsCurrent(WorkQueue::main());
+    auto queue = WorkQueue::create("com.apple.WebKit.Test.MainWorkQueueIsCurrent");
+    queue->dispatchSync([mainWorkQueue = Ref { WorkQueue::main() }] {
```

Comment:
I think you could :
* name it the way I described in slack so that the test suddenly doesn't just disappear on Release
* test both dispatchSync and dispatch, so that if dispatchSync is ever fixed to be similar to real dispatch, the test covers both
* test both negative case as you have here, as well as the positive case like I mentioned in Slack